### ISSUE TITLE
ExecuteResult isn't being called if IActionResult was returned from an action with object return typ

### ISSUE THREAD

**Issue by kekekeks** [aspnet/Mvc#5337]
```csharp
public object Index()
{
    return new JsonResult(new {test = 123});
}
```

Expected:
```json
{"test":123}
```

Actual:
```json
{"contentType":null,"serializerSettings":null,"statusCode":null,"value":{"test":123}}
```

This code has worked perfectly fine with RC1 and previous (classic) versions of ASP.NET MVC, so this is a regression.

**Comment by Eilon** [comment]
@rynowak I know we've talked about this before but I can't find a dup bug. Any idea where it might be? Or did we never file a bug?

**Comment by nil4** [comment]
Sounds like #4960

**Comment by Eilon** [comment]
@nil4 yes that's exactly it, thanks! Looks like we fixed it in 1.1.0.
